src/library/scala/collection/StringOps.scala

```diff
@@ -56,7 +56,7 @@ final class StringOps(private val s: String)
     *                `f` to each element of this String and collecting the results.
     */
   def map(f: Char => Char): String = {
-    val sb = new StringBuilder
+    val sb = new StringBuilder(length)
```

Comment:
I actually did that initially and then reverted thinking it was anti-scala

Reply:
It would be neat if the compiler did this automatically, for a local usage, to fall back from scala to java API if possible. There are cases (scala.js) where it's not clear what is portable or best. Also, if my handle were `isaacl`, I'd go with `iscala`.

Reply:
For now, let's add it back. We're stuck with the Scala version in `protected[this] def newSpecificBuilder`, but for these method-local versions I'd suggest avoiding the needless wrapper of `scala.StringBuilder`.